fs/dafsserver/dafsserver.cpp

```diff
@@ -132,21 +132,31 @@ static ISecureSocket *createSecureSocket(ISocket *sock, bool disableClientCertVe
         CriticalBlock b(secureContextCrit);
         if (!secureContextServer)
         {
-#ifdef _CONTAINERIZED
-            /* Connections are expected from 3rd parties via TLS,
-             * we do not expect them to provide a valid certificate for verification.
-             * Currently the server (this dafilesrv), will use either the "public" certificate issuer,
-             * unless it's visibility is "cluster" (meaning internal only)
-             */
+            if (isContainerized())
+            {
+                /* Connections are expected from 3rd parties via TLS,
+                * we do not expect them to provide a valid certificate for verification.
+                * Currently the server (this dafilesrv), will use either the "public" certificate issuer,
+                * unless it's visibility is "cluster" (meaning internal only)
+                */
 
-            const char *certScope = strsame("cluster", getComponentConfigSP()->queryProp("service/@visibility")) ? "local" : "public";
-            Owned<const ISyncedPropertyTree> info = getIssuerTlsSyncedConfig(certScope, nullptr, disableClientCertVerification);
-            if (!info || !info->isValid())
-                throw makeStringException(-1, "createSecureSocket() : missing MTLS configuration");
-            secureContextServer.setown(createSecureSocketContextSynced(info, ServerSocket));
-#else
-            secureContextServer.setown(createSecureSocketContextEx2(securitySettings.getSecureConfig(), ServerSocket));
-#endif
+                const char *certScope = strsame("cluster", getComponentConfigSP()->queryProp("service/@visibility")) ? "local" : "public";
+                Owned<const ISyncedPropertyTree> info = getIssuerTlsSyncedConfig(certScope, nullptr, disableClientCertVerification);
+                if (!info || !info->isValid())
+                    throw makeStringException(-1, "createSecureSocket() : missing MTLS configuration");
+                secureContextServer.setown(createSecureSocketContextSynced(info, ServerSocket));
+            }
+            else
+            {
+                IPropertyTree *cert = getComponentConfigSP()->getPropTree("cert");
+                if (cert)
+                {
+                    Owned<ISyncedPropertyTree> certSyncedWrapper = createSyncedPropertyTree(cert);
```

Comment:
minor: cert will be leaked - line 151 should be an owned instance or call queryPropTree

Reply:
whoops, should be owned, to avoid theoretical config change and invalidation of it before being used.
Will change.